src/alfasim_sdk/_internal/alfacase/plugin_alfacase_to_case.py

```diff
@@ -197,14 +197,31 @@ def load_plugin_data_structure(plugin_id: str) -> Optional[List[Type]]:
     """
     Obtain the models for a given plugin.
     """
+    import alfasim_sdk_plugins
+
     for candidate in get_plugin_module_candidates(plugin_id):
         if candidate.is_file():
+            # Update the "alfasim_sdk_plugins" namespace.
+            namespace_package_str = None
+            remove_namespace_package = False
+            namespace_package = candidate.parent / "alfasim_sdk_plugins"
+            if namespace_package.is_dir():
+                namespace_package_str = str(namespace_package.absolute())
+                if namespace_package_str not in alfasim_sdk_plugins.__path__:
+                    alfasim_sdk_plugins.__path__.append(namespace_package_str)
+                    remove_namespace_package = True
+
             module = import_module(candidate)
             if hasattr(module, "alfasim_get_data_model_type"):
                 models = module.alfasim_get_data_model_type()
                 if all(hasattr(m, "_alfasim_metadata") for m in models):
                     return models
 
+            # Cleanup "alfasim_sdk_plugins" namespace.
+            # When a valid plugin is found leave the namespace package unchanged.
+            if remove_namespace_package:
+                alfasim_sdk_plugins.__path__.remove(namespace_package_str)
+
```

Comment:
Just to confirm: Is it expected to have both conditions satisfied, and If so it's ok to not clean the "alfasim_sdk_plugins" namespace in this case?

Reply:
Yes:

- already is there: when alfasim is exporting an `.alfacase` this code will be triggered to export the relevant user plugin data. In this case the user plugin's package will already be accessible and should remain available after;
- don't cleanup: if the returned module does some lazy imports (local to a function) that import will fail if we cleanup after finding the plugin spec module;
- why check before add: I don't like the idea of having multiple equal entries, or invalid entries;